tools/tpm2_load.c

```diff
@@ -54,6 +89,184 @@ static tool_rc load(ESYS_CONTEXT *ectx) {
+    .size = 0,
+};
+
+static const TPML_PCR_SELECTION allCreationPCR = {
+    .count = 0,
+};
+
+/*static const TPM2B_PUBLIC primaryRsaTemplate = {
+    .publicArea = {
+        .type = TPM2_ALG_RSA,
+        .nameAlg = TPM2_ALG_SHA256,
+        .objectAttributes = (TPMA_OBJECT_USERWITHAUTH |
+                             TPMA_OBJECT_RESTRICTED |
+                             TPMA_OBJECT_DECRYPT |
+                             TPMA_OBJECT_NODA |
+                             TPMA_OBJECT_FIXEDTPM |
+                             TPMA_OBJECT_FIXEDPARENT |
+                             TPMA_OBJECT_SENSITIVEDATAORIGIN),
+        .authPolicy = {
+             .size = 0,
+         },
+        .parameters.rsaDetail = {
+             .symmetric = {
+                 .algorithm = TPM2_ALG_AES,
+                 .keyBits.aes = 128,
+                 .mode.aes = TPM2_ALG_CFB,
+              },
+             .scheme = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+             .keyBits = 2048,
+             .exponent = 0,
+         },
+        .unique.rsa = {
+             .size = 0,
+         }
+     }
+};*/
+
+static const TPM2B_PUBLIC primaryEccTemplate = {
+    .publicArea = {
+        .type = TPM2_ALG_ECC,
+        .nameAlg = TPM2_ALG_SHA256,
+        .objectAttributes = (TPMA_OBJECT_USERWITHAUTH |
+                             TPMA_OBJECT_RESTRICTED |
+                             TPMA_OBJECT_DECRYPT |
+                             TPMA_OBJECT_NODA |
+                             TPMA_OBJECT_FIXEDTPM |
+                             TPMA_OBJECT_FIXEDPARENT |
+                             TPMA_OBJECT_SENSITIVEDATAORIGIN),
+        .authPolicy = {
+             .size = 0,
+         },
+        .parameters.eccDetail = {
+             .symmetric = {
+                 .algorithm = TPM2_ALG_AES,
+                 .keyBits.aes = 128,
+                 .mode.aes = TPM2_ALG_CFB,
+              },
+             .scheme = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+             .curveID = TPM2_ECC_NIST_P256,
+             .kdf = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+         },
+        .unique.ecc = {
+             .x.size = 0,
+             .y.size = 0
+         }
+     }
+};
+
+void print_parent_name(ESYS_CONTEXT *esys_ctx, ESYS_TR *parent) {
+    TPM2B_NAME *name = 0;
+    tool_rc rc = tpm2_tr_get_name(esys_ctx, *parent, &name);
+    if (rc != tool_rc_success) {
+        printf("Unable to get parent name\n");
+    }
+    printf("Name Before: ");
+    for (int i = 0; i < name->size; i++) {
+        printf("%x\n", name->name[i]);
+    }
+    printf("\n");
+}
+
+tool_rc setup_primary(ESYS_CONTEXT *esys_ctx, ESYS_TR *parent) {
+    TSS2_RC r;
+    const TPM2B_PUBLIC *primaryTemplate = NULL;
+    //ESYS_TR primary;
+
+    primaryTemplate = &primaryEccTemplate;
+
+    if(!primaryTemplate) {
```

Comment:
```suggestion
    if(!primary_template) {
```
no camel casing please